trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py

```diff
@@ -0,0 +1,194 @@
+        self.outputs.specific_impulse                 = 0.0
+        self.outputs.non_dimensional_thrust           = 0.0
+        self.outputs.mass_flow_rate                   = 0.0
+        self.design_thrust                            = 0.0
+        self.ISP_design                               = 0.0
+
+    def compute(self,conditions):
+        """Computes thrust and other properties as below.
+
+        Assumptions:
+        Perfect gas
+
+        Source:
+        https://web.stanford.edu/~cantwell/AA283_Course_Material/AA283_Course_Notes/
+
+        Inputs:
+        conditions.freestream.
+          pressure                           [Pa]
+          temperature                        [K]
+          gravity                            [-]
+        conditions.throttle                  [-]
+        self.inputs.
+          propellant_mass_rate               [kg/s]
+          combustion_pressure                [Pa]
+          expansion_ratio                    [-]
+          area_throat                        [m^2]
+          exhaust_velocity                   [m/s]
+          number_of_engines                  [-]
+
+        Outputs:
+        self.outputs.
+          thrust                             [N]
+          propellant_mass_flow_rate          [kg/s]
+          power                              [W]
+          specific_impulse                   [s]
+          exhuast_velocity                   [m/s]
+
+        Properties Used:
+        
+        """  
+        
+        #--Unpack the values--
+
+        # unpacking from conditions
+        p0               = conditions.freestream.pressure  
+        g0               = conditions.freestream.gravity
+        throttle         = conditions.propulsion.throttle
+        
+        # unpacking from inputs
+        Pt_combustion    = self.inputs.combustion_pressure
+        pe               = self.inputs.static_pressure         
+        expansion_ratio  = self.inputs.expansion_ratio
+        area_throat      = self.inputs.area_throat
+        Me               = self.inputs.mach_number
+        exhaust_velocity = self.inputs.exhaust_velocity
+        num_eng          = self.inputs.number_of_engines 
+        gamma            = self.inputs.isentropic_expansion_factor
+        rho              = self.inputs.density
+              
+        #--Computing the thrust coefficient and effective exhaust velocity--
+        gmm = gamma*Me*Me
+        C  = exhaust_velocity*(1.+1./(gmm)*(1.-p0/pe))
+        
+        #--Computing specific impulse
+        Isp           = C/g0        
+        
+        #--Computing the propellant/exhaust mass rate
+        Ae            = area_throat*expansion_ratio
+        mdot_temp     = throttle*rho*exhaust_velocity*Ae
+        
+        #--Computing Dimensional Thrust
+        thrust        = num_eng*mdot_temp*C
+        
+        #--Make mdot size of thrust
+        mdot               = mdot_temp 
+        mdot[mdot<0.0]     = 0.0 
+        thrust[thrust<0.0] = 0.0
```

Comment:
lol, come on no negative thrust!...

actually, you may find removing this could help convergence.

Reply:
Should it be removed as it stands currently?   I will try comment these in the X1 case (im not hopeful) hahah.

Reply:
doesn't help x1 convergence